contrib/google.golang.org/grpc/grpc_test.go

```diff
@@ -30,42 +29,73 @@ func TestClient(t *testing.T) {
 	defer rig.Close()
 	client := rig.client
 
-	span, ctx := tracer.StartSpanFromContext(context.Background(), "a", tracer.ServiceName("b"), tracer.ResourceName("c"))
-
-	resp, err := client.Ping(ctx, &FixtureRequest{Name: "pass"})
-	assert.Nil(err)
-	span.Finish()
-	assert.Equal(resp.Message, "passed")
-
-	spans := mt.FinishedSpans()
-	assert.Len(spans, 3)
+	testcase := map[string]struct {
+		message     string
+		error       bool
+		wantMessage string
+		wantCode    codes.Code
+	}{
+		"OK": {
+			"pass",
+			false,
+			"passed",
+			codes.OK,
+		},
+		"Invalid": {
+			"invalid",
+			true,
+			"",
+			codes.InvalidArgument,
+		},
+	}
 
-	var serverSpan, clientSpan, rootSpan mocktracer.Span
+	for name, tt := range testcase {
+		t.Run(name, func(t *testing.T) {
+			mt := mocktracer.Start()
+			defer mt.Stop()
 
-	for _, s := range spans {
-		// order of traces in buffer is not garanteed
-		switch s.OperationName() {
-		case "grpc.server":
-			serverSpan = s
-		case "grpc.client":
-			clientSpan = s
-		case "a":
-			rootSpan = s
-		}
-	}
+			span, ctx := tracer.StartSpanFromContext(context.Background(), "a", tracer.ServiceName("b"), tracer.ResourceName("c"))
 
-	assert.NotNil(serverSpan)
-	assert.NotNil(clientSpan)
-	assert.NotNil(rootSpan)
+			resp, err := client.Ping(ctx, &FixtureRequest{Name: tt.message})
+			span.Finish()
+			if tt.error {
+				assert.Error(err)
+			} else {
+				assert.Nil(err)
```

Comment:
You can use `assert.NoError(err)`. It's not necessarily more correct, but I guess we should make use of it.